.circleci/config.yml

```diff
@@ -23,7 +23,7 @@ jobs:
             - ~/.m2
       - run:
           name: "Running build"
-          command: ./mvnw -s .settings.xml clean org.jacoco:jacoco-maven-plugin:prepare-agent install -U -P sonar -nsu --batch-mode -Dmaven.test.redirectTestOutputToFile=true -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+          command: ./mvnw -s .settings.xml clean org.jacoco:jacoco-maven-plugin:prepare-agent install -U -P sonar,withoutDockerTests -nsu --batch-mode -Dmaven.test.redirectTestOutputToFile=true -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
```

Comment:
I think I agree with @jonatan-ivanov that these could run with a `machine` type as they do in gateway.

https://github.com/spring-cloud/spring-cloud-gateway/blob/main/.circleci/config.yml#L4

Reply:
I agree changing the CircleCI build is the ideal scenario.  I tried that with [an early commit](https://github.com/spring-cloud/spring-cloud-config/pull/1971/commits/8dccda6bb941e1832a41070d265fd5da2acbc293) but it caused a lot of errors that were unrelated to this change of introducing Testcontainers.  If I remember correctly I think it dealt with configuring the Java Cryptography Architecture (JCA) on the CircleCI environment.  
I was planning to get this code ready to go first and then revisit changing the CircleCI build with the team later.  But we can certainly tackle that now if that's the preference.

Reply:
I see two issues, both are pretty weird:
```
Cannot mkdir: No such file or directory tar: home/appuser
```
I'm not sure why is this happening, Micrometer is doing the same (writing `~/.gradle` on a `machine`).


This one scares me a little:
```
Caused by: java.security.InvalidKeyException: Illegal key size
```
One possible cause of this is not having the "Java Cryptography Extension (JCE) Unlimited Strength Jurisdiction Policy" files installed. The problem with this is that Java 8 has these files since [Java 8u161](https://www.oracle.com/java/technologies/javase/8u161-relnotes.html#JDK-8170157) (4 years), previously you needed to install them. This makes me wonder if `machine: true` will give us some ancient image that has a very old version of Java 8.

Can we try what happens when we [set the machine image](https://circleci.com/docs/2.0/executor-types/#using-machine) to the [latest ubuntu that circleci has](https://circleci.com/docs/2.0/configuration-reference/#available-machine-images)?

Reply:
Great idea @jonatan-ivanov !  Specifying the machine image got rid of the errors around the `Illegal key size`.

However to get the build to work I still had to remove the `restore_cache` step.  In a previous build this error came back:

`Cannot mkdir: Permission denied tar: home/appuser/.m2`

Any ideas how to restore the cache?  I've been comparing it to other Spring Cloud projects and can't find any practical difference.

Reply:
I had the same issue when transitioning gateway to machine. Basically, remove the cache, then add it back. We can do that after it is merged.